e2e/dractions/retry.go

```diff
@@ -143,3 +146,72 @@ func waitDRPCProgression(
 		}
 	}
 }
+
+func addDRPCAnnotation(ctx types.TestContext, drpcName string) error {
+	log := ctx.Logger()
+
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		drpc, err := getDRPC(ctx, ctx.ManagementNamespace(), drpcName)
+		if err != nil {
+			return err
+		}
+
+		if drpc.Annotations == nil {
+			drpc.Annotations = make(map[string]string)
+		}
+
+		drpc.Annotations[drpcDoNotDeletePVC] = "true"
+
+		if err := updateDRPC(ctx, drpc); err != nil {
+			return err
+		}
+
+		log.Debugf("Annotated drpc \"%s/%s\" with \"%s: %s\" in cluster %q",
+			ctx.ManagementNamespace(), drpcName, drpcDoNotDeletePVC,
+			drpc.Annotations[drpcDoNotDeletePVC], ctx.Env().Hub.Name)
+
+		return nil
+	})
+}
+
+func waitForDRPCAnnotationPropagation(ctx types.TestContext, drpcName string) error {
+	log := ctx.Logger()
+	start := time.Now()
+
+	log.Debugf("Waiting until drpc \"%s/%s\" annotation %q is propagated to primary vrg",
+		ctx.ManagementNamespace(), drpcName, drpcDoNotDeletePVC)
+
+	for {
+		drpc, err := getDRPC(ctx, ctx.ManagementNamespace(), drpcName)
+		if err != nil {
+			return err
+		}
+
+		preferredCluster, err := ctx.Env().GetCluster(drpc.Spec.PreferredCluster)
+		if err != nil {
+			return err
+		}
```

Comment:
We do this at the end of some dr action, so the preferred cluster should not change during the operation. Lets move these steps out of the loop. The loop should get only the vrg.